src/core/job.c

```diff
@@ -598,40 +598,56 @@ static const char* job_done_message_format(Unit *u, JobType t, JobResult result)
         static const char* const generic_finished_verify_active_job[_JOB_RESULT_MAX] = {
                 [JOB_SKIPPED]     = "%s is inactive.",
         };
-        const char *format;
 
         assert(u);
         assert(t >= 0);
         assert(t < _JOB_TYPE_MAX);
 
         /* Show condition check message if the job did not actually do anything due to failed condition. */
-        if (t == JOB_START && result == JOB_DONE && !u->condition_result)
-                return "Condition check resulted in %s being skipped.";
+        if (t == JOB_START && result == JOB_DONE && !u->condition_result) {
+                Condition *c;
+
+                LIST_FOREACH(conditions, c, u->conditions)
+                        if (c->result != CONDITION_SUCCEEDED)
+                                break;
+
+                assert(c);
```

Comment:
I'm not entirely sure if this is safe. If condition_result indicates a condition failed, will there always be a condition whose result will not be CONDITION_SUCCEEDED?

Reply:
CI gave the answer, turns out this does happen, replaced with NULL check.